src/fileformat/file_format/pe/pe_format.cpp

```diff
@@ -3350,5 +3409,299 @@ const VisualBasicInfo* PeFormat::getVisualBasicInfo() const
+		std::string msgName = (res->getNameId(nameId)) ? numToStr(nameId) : "<unknown>";
+
+		// scan for resource size over 100MB
+		if (res->getSizeInFile() >= 100000000UL)
+		{
+			anomalies.emplace_back("largeRes", "Resource " + msgName + " has size over 100MB");
+		}
+
+		// scan for resource stretched over multiple sections
+		unsigned long long resAddr;
+		if (getAddressFromOffset(resAddr, res->getOffset()) &&
+			isObjectStretchedOverSections(resAddr, res->getSizeInFile()))
+		{
+			anomalies.emplace_back("stretchedRes", "Resource " + msgName + " is stretched over multiple sections");
+		}
+	}
+}
+
+/**
+ * Scan for import anomalies
+ */
+void PeFormat::scanForImportAnomalies()
+{
+	// scan for import stretched over multiple sections
+	for(const auto &impRange : formatParser->getImportDirectoryOccupiedAddresses())
+	{
+		unsigned long long impAddr;
+		if (getAddressFromOffset(impAddr, impRange.getStart()) &&
+			isObjectStretchedOverSections(impAddr, impRange.getSize()))
+		{
+			std::string msgName;
+			auto imp = getImport(impAddr);
+			if (!imp)
+			{
+				msgName = "<unknown>";
+			}
+			else
+			{
+				if (imp->hasEmptyName())
+				{
+					unsigned long long ordNum;
+					if (!imp->getOrdinalNumber(ordNum))
+					{
+						msgName = "<unknown>";
+					}
+					else
+					{
+						msgName = numToStr(ordNum);
+					}
+
+				}
+				else
+				{
+					msgName = imp->getName();
+				}
+			}
+
+			anomalies.emplace_back("stretchedImp", "Import " + msgName + " is stretched over multiple sections");
+		}
+	}
+}
+
+/**
+ * Scan for export anomalies
+ */
+void PeFormat::scanForExportAnomalies()
+{
+	// scan for export stretched over multiple sections
+	for(const auto &expRange : formatParser->getExportDirectoryOccupiedAddresses())
+	{
+		unsigned long long expAddr;
+		if (getAddressFromOffset(expAddr, expRange.getStart()) &&
+			isObjectStretchedOverSections(expAddr, expRange.getSize()))
+		{
+			std::string msgName;
+			auto exp = getExport(expAddr);
+			if (!exp)
+			{
+				msgName = "<unknown>";
+			}
+			else
+			{
+				if (exp->hasEmptyName())
+				{
+					unsigned long long ordNum;
+					if (!exp->getOrdinalNumber(ordNum))
+					{
+						msgName = "<unknown>";
+					}
+					else
+					{
+						msgName = numToStr(ordNum);
+					}
+
+				}
+				else
+				{
+					msgName = exp->getName();
+				}
+			}
+
+			anomalies.emplace_back("stretchedExp", "Export " + msgName + " is stretched over multiple sections");
+		}
+	}
+}
+
+/**
+ * Scan for optional header anomalies
+ */
+void PeFormat::scanForOptHeaderAnomalies()
+{
+	// scan for missalignment
+	if (!formatParser->isIsSizeOfHeaderMultipleOfFileAlignment())
+	{
+		anomalies.emplace_back("sizeOfHeadersNotAligned", "SizeOfHeaders is not aligned to multiple of FileAlignment");
+	}
+}
+
+/**
+ * Scan for obsolete file characteristics
+ */
+void PeFormat::scanForObsoleteCharacteristics()
+{
+	std::size_t flags = getFileFlags();	
+	std::size_t obsoleteFlags = PELIB_IMAGE_FILE_LINE_NUMS_STRIPPED |
```

Comment:
Where did you take this anomaly from? I am just curious whether each of this flag is really obsolete. https://docs.microsoft.com/en-us/windows/desktop/api/winnt/ns-winnt-_image_file_header says only few of the ones you 've listed are.

Reply:
I have adopted it directly from source codes of [PortEx](https://github.com/katjahahn/PortEx). The link you posted seems to ommit the fact that the flags are indeed obsolete (see [flags documentation](https://docs.microsoft.com/en-us/windows/desktop/debug/pe-format#characteristics) referenced by [PE documentation](https://docs.microsoft.com/en-us/windows/desktop/debug/pe-format#coff-file-header-object-and-image) ). I checked whether the `Obsolete Flag` anomaly pops on binary I have created with VB6 compiler and it does. Consider neglecting the flags that turned obsolete "recently".